physical-based-animations.go

```diff
@@ -90,53 +91,108 @@ func run() {
+
+	gui.NewButton(pauseButton)
+
+	stopButton := Button{
+		position:     pixel.V(10, 190),
+		croppingArea: pixel.R(0, 0, 80, 80),
+		bounds:       pixel.R(0, 0, 80, 80),
+		onClick:      HandleStopClick,
+	}
+
+	gui.NewButton(stopButton)
+
+	cam := pixel.IM.Scaled(camPos, 1.0).Moved(win.Bounds().Center().Sub(camPos))
+
+	win.SetMatrix(cam)
+
+	gui.SetMatrix(cam)
+	gui.BindState(&state)
+	gui.MainLoop()
+	gui.Draw()
+
 	for !win.Closed() {
-		dt := time.Since(last).Seconds()
-		last = time.Now()
-		timeElapsed += dt
 
-		cam := pixel.IM.Scaled(camPos, 1.0).Moved(win.Bounds().Center().Sub(camPos))
-		win.SetMatrix(cam)
-		win.Clear(colornames.Whitesmoke)
+		win.Update()
 
 		batch.Clear()
-		updateParticles(particleSystem.particles, batch, dt, cam)
-
-		batch.Draw(win)
-
-		for timeElapsed > timeForOneParticle {
-			pos := particleSystem.position
-			angle := (rand.Float64() - 0.5) * (particleSystem.angle * (math.Pi / 180))
-			speed := pixel.V(0, startSpeed).Rotated(angle)
-			nextPost := pos.Add(speed.Scaled(PixelsPerMeter).Scaled(prevDt)).Add(Gravity.Scaled(PixelsPerMeter).Scaled(prevDt * prevDt * 0.5))
-			// fmt.Printf("pos: %f %f	prev: %f %f\n", pos.X, pos.Y, prevPos.X, prevPos.Y)
-			particle := Particle{
-				position:     pos,
-				nextPosition: nextPost,
-				speed:        speed,
-				prevDt:       prevDt,
-				sprite:       *particleSprite,
-				lifespan:     100.0,
-				alive:        0.0,
-			}
-			particleSystem.particles = append(particleSystem.particles, particle)
-			timeElapsed = timeElapsed - timeForOneParticle
-		}
 
-		win.Update()
+		if !gui.GetState().paused && !gui.GetState().stopped {
+			dt := time.Since(last).Seconds()
+			last = time.Now()
+			timeElapsed += dt
+
+			updateParticles(particleSystem.particles, batch, dt, cam)
+
+			win.Clear(colornames.Whitesmoke)
+			gui.batch.Draw(win)
+			batch.Draw(win)
+
+			for timeElapsed > timeForOneParticle {
+				pos := particleSystem.position
+				angle := (rand.Float64() - 0.5) * (particleSystem.angle * (math.Pi / 180))
+				speed := pixel.V(0, startSpeed).Rotated(angle)
+				nextPost := pos.Add(speed.Scaled(PixelsPerMeter).Scaled(prevDt)).Add(
+					Gravity.Scaled(PixelsPerMeter).Scaled(prevDt * prevDt * 0.5))
+				// fmt.Printf("pos: %f %f	prev: %f %f\n", pos.X, pos.Y, prevPos.X, prevPos.Y)
+				particle := Particle{
+					position:     pos,
+					nextPosition: nextPost,
+					speed:        speed,
+					prevDt:       prevDt,
+					sprite:       *particleSprite,
+					lifespan:     100.0,
+					alive:        0.0,
+				}
+				particleSystem.particles = append(particleSystem.particles, particle)
+				timeElapsed = timeElapsed - timeForOneParticle
+			}
 
-		frames++
-		select {
-		case <-second:
-			particleSystem.KillOldParticles(
-				win.Bounds().Min.X,
-				win.Bounds().Max.X,
-				win.Bounds().Min.Y,
-			)
-			win.SetTitle(fmt.Sprintf("%s | FPS: %d | particles %d", cfg.Title, frames,
-				len(particleSystem.particles)))
-			frames = 0
-		default:
+			frames++
+			select {
```

Comment:
Also would it be possible to have this select outside of the `running` condition? As it doesn't affect rendering but it would help to see for example the particle count and fps